ml/Host.cc

```diff
@@ -401,6 +421,32 @@ void DetectableHost::detectOnce() {
     updateEventsChart(getRH(), P, ResetBitCounter, NewAnomalyEvent);
     updateTrainingChart(getRH(), TotalTrainingDuration * 1000.0, MaxTrainingDuration * 1000.0);
 
+    /*code snippet to keep account of the count of the anomalous values of each dimension*/
+    /*...in the anomaly-precentage period configured by (Cfg.SaveAnomalyPercentageEvery)*/
+    if(AnomalyBitCounterWindow == 0) {
+        /*one period is completed, save in the DB the vector that holds the values of the percentages 
+        (of the set anomaly bits) for each dimension*/
+        nlohmann::json JsonResult = DimsAnomalyRate;
+
+        time_t Before = now_realtime_sec();
+        time_t After = Before - (Cfg.SaveAnomalyPercentageEvery * updateEvery());
+        
+        //DB.insertAnomalyRateInfo(getUUID(), After, Before, JsonResult.dump(4));
+        /*for(size_t i = 0; i < DimsAnomalyRate.size(); i++) {
+            DB.insertAnomalyRateInfo(getUUID(), DimsAnomalyRate[i].second, After, Before, DimsAnomalyRate[i].first);
+        }*/
+        for(std::vector<std::pair<double, std::string>>::iterator it = DimsAnomalyRate.begin(); it != DimsAnomalyRate.end(); ++it) {
+            DB.insertAnomalyRateInfo(getUUID(), it->second, After, Before, it->first);
+        }
+        //DB.insertAnomalyRatesInfo(getUUID(), After, Before, JsonResult.dump(4));
```

Comment:
Will you follow this approach? I'm asking because the code is commented out.

Personally, I like the simplicity of this approach. However, from a performance perspective it's not the optimal one because we are serializing the `DimsAnomalyRate` vector into JSON and deserializing it in the correspondent sqlite statement.

Can you profile it and see what is the cost in terms of CPU/memory usage?

Reply:
I am aiming at making josn array instead of vector in the first place.